stochrare/dynamics/diffusion.py

```diff
@@ -167,8 +200,14 @@ def trajectory(self, x0, t0, **kwargs):
         if 'brownian_path' in kwargs:
             tw, w = kwargs.pop('brownian_path')
             dw = np.diff(w, axis=0)
+            deltat = tw[1]-tw[0]
+            ratio = int(np.rint(dt/deltat)) # Both int and rint needed here ?
+            dw = dw[:((num-1)*ratio)] # Trim noise vector if sequence w too long
         else:
+            deltat = kwargs.pop('deltat', dt)
+            ratio = int(np.rint(dt/deltat))
             dw = np.random.normal(0, np.sqrt(dt), size=(num-1, dim))
+        dw = self._integrate_brownian_path(dw, num, dim, ratio)
```

Comment:
It might not affect performance, but perhaps we could call `_integrate_brownian_path` only if `deltat != dt` (or `ratio != 1`).